sdk/ai/azure-ai-projects/azure/ai/projects/telemetry/agents/_ai_agents_instrumentor.py

```diff
@@ -0,0 +1,1345 @@
+        
+        with self.start_submit_tool_outputs_span(
+            project_name=project_name,
+            thread_id=thread_id,
+            run_id=run_id,
+            tool_outputs=tool_outputs,
+            event_handler=event_handler) as span:
+            try:
+                if stream:
+                    kwargs['event_handler'] = self.wrap_handler(event_handler, span)
+
+                result = function(*args, **kwargs)
+                self.set_end_run(span, result)
+            except Exception as exc:
+                # Set the span status to error
+                if isinstance(span.span_instance, Span):  # pyright: ignore [reportPossiblyUnboundVariable]
+                    span.span_instance.set_status(
+                        StatusCode.ERROR,  # pyright: ignore [reportPossiblyUnboundVariable]
+                        description=str(exc),
+                    )
+                module = getattr(exc, "__module__", "")
+                module = module if module != "builtins" else ""
+                error_type = f"{module}.{type(exc).__name__}" if module else type(exc).__name__
+                self._set_attributes(span, ("error.type", error_type))
+                raise
+
+        return result
+
+    async def trace_submit_tool_outputs_async(self, stream, function, *args, **kwargs):
+        project_name = args[0]._config.project_name
+        thread_id = kwargs.get("thread_id")
+        run_id = kwargs.get("run_id")
+        tool_outputs = kwargs.get("tool_outputs")
+        event_handler = kwargs.get("event_handler")
+        
+        with self.start_submit_tool_outputs_span(
+            project_name=project_name,
+            thread_id=thread_id,
+            run_id=run_id,
+            tool_outputs=tool_outputs,
+            event_handler=event_handler) as span:
+            try:
+                if stream:
+                    kwargs['event_handler'] = self.wrap_handler(event_handler, span)
+
+                result = await function(*args, **kwargs)
+                self.set_end_run(span, result)
+            except Exception as exc:
+                # Set the span status to error
+                if isinstance(span.span_instance, Span):  # pyright: ignore [reportPossiblyUnboundVariable]
+                    span.span_instance.set_status(
+                        StatusCode.ERROR,  # pyright: ignore [reportPossiblyUnboundVariable]
+                        description=str(exc),
+                    )
+                module = getattr(exc, "__module__", "")
+                module = module if module != "builtins" else ""
+                error_type = f"{module}.{type(exc).__name__}" if module else type(exc).__name__
+                self._set_attributes(span, ("error.type", error_type))
+                raise
+
+        return result
+
+    def trace_handle_submit_tool_outputs(self, function, *args, **kwargs):
+        event_handler = kwargs.get("event_handler")
+        span = getattr(event_handler, "span", None)
```

Comment:
if I run `.\samples\agents\sample_agents_stream_eventhandler_with_toolset.py` this line throws with 

```
  File "C:\Users\limolkova\AppData\Local\Programs\Python\Python312\Lib\site-packages\azure\ai\projects\tracing\agents\_ai_agents_instrumentor.py", line 846, in trace_handle_submit_tool_outputs
    with span.change_context(span.span_instance):
         ^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'change_context'
```

it seems the original event handler is passed and not the instrumented one

Reply:
ah, the `event_handler` is None, so probably something does not pass it

Reply:
i made a fix for this, for some reason the event_handler ends up in args. i need to investigate this more to understand it better, but at least with this change the code work